### PR TITLE
Makes welder code better*

### PR DESCRIPTION
**Pull request by Trilbyspaceclone**
## About The Pull Request

Makes welder code just be better
Buffs Brass welder to reload itself faster, thats it

## Why It's Good For The Game

Brass welders are cult and magic should, be somewhat better imo, legit dosnt do to much in terms of active gameplay
Code wise - easier to read and more cleanly imo runs
## Changelog
:cl:
balance: Brass welders are 50% faster at refueling
code: redoes self fueling welders in the code to be less speggie
/:cl:

### CODE REVIEW

File: code/game/objects/items/tools/weldingtool.dm [hunk 1]
```diff
@@ -21,6 +21,10 @@
 	armor = list("melee" = 0, "bullet" = 0, "laser" = 0, "energy" = 0, "bomb" = 0, "bio" = 0, "rad" = 0, "fire" = 100, "acid" = 30)
 	resistance_flags = FIRE_PROOF
 
+	var/self_fueling = FALSE //Do we refill areself or not
```
Comment (jjpark-kb) [inline code/game/objects/items/tools/weldingtool.dm]:
```suggestion
	var/self_fueling = FALSE //Do we refill ourselves or not
```

File: code/game/objects/items/tools/weldingtool.dm [hunk 2]
```diff
@@ -82,12 +86,18 @@
 	//This is to start fires. process() is only called if the welder is on.
 	open_flame()
 
+	//This handles refuelung, its looking at how much fuel the tool has, and comparing it to the how much it holds
+	//This then looks if the refuel tick has come based on world time.
+	//Then looks if we refuel areselfs are not.
```
Comment (jjpark-kb) [inline code/game/objects/items/tools/weldingtool.dm]:
```suggestion
	//Then looks if we refuel ourselves or not.
```

File: code/game/objects/items/tools/weldingtool.dm [hunk 3]
```diff
@@ -82,12 +86,18 @@
 	//This is to start fires. process() is only called if the welder is on.
 	open_flame()
 
+	//This handles refuelung, its looking at how much fuel the tool has, and comparing it to the how much it holds
```
Comment (jjpark-kb) [inline code/game/objects/items/tools/weldingtool.dm]:
```suggestion
	//This handles refueling. Its looking at how much fuel the tool has and comparing that to how much it holds
```

File: code/game/objects/items/tools/weldingtool.dm [hunk 4]
```diff
@@ -353,53 +348,57 @@
...
 	can_off_process = 1
 	light_intensity = 1
 	toolspeed = 0.5
-	var/nextrefueltick = 0
 
 /obj/item/weldingtool/experimental/brass
 	name = "brass welding tool"
 	desc = "A brass welder that seems to constantly refuel itself. It is faintly warm to the touch."
 	resistance_flags = FIRE_PROOF | ACID_PROOF
+	next_fueling = 5 //makes welding fuel faster then experimental ones
```
Comment (jjpark-kb) [inline code/game/objects/items/tools/weldingtool.dm]:
```suggestion
	next_fueling = 5 //makes welding fuel faster than experimental ones
```

File: code/game/objects/items/tools/weldingtool.dm [hunk 5]
```diff
@@ -21,6 +21,10 @@
 	armor = list("melee" = 0, "bullet" = 0, "laser" = 0, "energy" = 0, "bomb" = 0, "bio" = 0, "rad" = 0, "fire" = 100, "acid" = 30)
 	resistance_flags = FIRE_PROOF
 
+	var/self_fueling = FALSE //Do we refill areself or not
+	var/next_fueling = 10 //How long do we wait before refilling
```
Comment (jjpark-kb) [inline code/game/objects/items/tools/weldingtool.dm]:
```suggestion
	var/next_fueling = 10 //How long we wait before refilling
```

**Review by silicons — CHANGES_REQUESTED** [review]
just do a variable for amount of fuel to regenerate per cycle

also this doesn't take into account that process() isn't going to be called if the welder isn't turned on

**Comment by Trilbyspaceclone** [comment]
@silicons done I guess

**Pull request merged**